pkg/apis/cluster/types.go

```diff
@@ -104,6 +125,101 @@ type ClusterSpec struct {
+	// When grade greater than or equal to 2, each default model's cpu quota is [2^(grade-1), 2^grade), 2 <= grade <= 7
+	// Each default model's memory quota is [2^(grade + 2), 2^(grade + 3)), 2 <= grade <= 7
+	// E.g. grade 0 likes this:
+	// - grade: 0
+	//   ranges:
+	//   - name: "cpu"
+	//     min: 0 C
+	//     max: 1 C
+	//   - name: "memory"
+	//     min: 0 GB
+	//     max: 4 GB
+	//
+	// - grade: 1
+	//   ranges:
+	//   - name: "cpu"
+	//     min: 1 C
+	//     max: 2 C
+	//   - name: "memory"
+	//     min: 4 GB
+	//     max: 16 GB
+	//
+	// - grade: 2
+	//   ranges:
+	//   - name: "cpu"
+	//     min: 2 C
+	//     max: 4 C
+	//   - name: "memory"
+	//     min: 16 GB
+	//     max: 32 GB
+	//
+	// - grade: 7
+	//   range:
+	//   - name: "cpu"
+	//     min: 64 C
+	//     max: 128 C
+	//   - name: "memory"
+	//     min: 512 GB
+	//     max: 1024 GB
+	//
+	// grade 8, the last one likes below. No matter what Max value you pass,
+	// the meaning of Max value in this grade is infinite. You can pass any number greater than Min value.
+	// - grade: 8
+	//   range:
+	//   - name: "cpu"
+	//     min: 128 C
+	//     max: MAXINT
+	//   - name: "memory"
+	//     min: 1024 GB
+	//     max: MAXINT
+	//
+	// +optional
+	ResourceModels []ResourceModel
+}
+
+// ResourceModel describes the modeling that you want to statistics.
+type ResourceModel struct {
+	// Grade is the index for the resource modeling.
+	// +optional
+	Grade int
+
+	// Ranges describes the resource quota ranges.
+	// +optional
+	Ranges []ResourceModelRange
+}
+
+// ResourceModelRange describes the detail of each modeling quota that ranges from min to max.
+// Please pay attention, by default, the value of min can be inclusive, and the value of max cannot be inclusive.
+// E.g. in an interval, min = 2, max =10 is set, which means the interval [2,10).
+// This rule ensure that all intervals have the same meaning. If the last interval is +∞,
+// it is definitely unreachable. Therefore, we define the right interval as the open interval.
+// For a valid interval, the value on the right is greater than the value on the left,
+// in other words, max must be greater than min.
+type ResourceModelRange struct {
+	// Name is the name for the resource that you want to categorize.
+	// +optional
+	Name ResourceName
+
+	// Min is the minimum amount of this resource represented by resource name.
+	// +optional
+	Min resource.Quantity
+
+	// Max is the maximum amount of this resource represented by resource name.
+	// Special Instructions, for the last ResourceModelRange, which no matter what Max value you pass,
+	// the meaning is infinite. Because for the last item,
+	// any ResourceModelRange's quota larger than Min will be classified to the last one.
+	// Of course, the value of the Max field is always greater than the value of the Min field.
+	// It should be true in any case.
+	// +optional
+	Max resource.Quantity
```

Comment:
> Special Instructions, for the last ResourceModelRange, which no matter what Max value you pass,
> the meaning is infinite. Because for the last item
Shall we make the last grade ignore user-input max value, or, add a computed last/largest grade according to last/largest user-input max value?

Reply:
My current implementation is that whatever value the last max is, those greater than min will be placed in the last group. This ensures that there will be no missing ResourceModelRange. Do you think my solution is reasonable?

Reply:
I agree with the implementation behavior, I guess what @kevin-wangzefeng mean is we should clearly note the behavior on API comments.

Reply:
By the way, the behavior description should be added to `ResourceModels`(.spec.ResourceModels), not on `ResourceModelRange` which just represents the range. How we use/treat ranges is decided at `ResourceModels`.